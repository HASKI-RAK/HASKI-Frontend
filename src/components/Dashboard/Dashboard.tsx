import { DefaultButton as Button } from "@common/components";
import { UserState, useUserStore } from "@services/UserStore";
export const Dashboard = ({
  userState = {
    user: useUserStore((state) => state.user),
    increaseUserId: useUserStore((state) => state.increaseUserId),
  },
}: DashboardProps) => (
  <>
    <div>
      Hello {userState.user?.firstName} {userState.user?.surName} with ID:
      {userState.user?.id}
    </div>
    <Button
      variant="contained"
      color="primary"
      onClick={userState.increaseUserId}
    >
      one up
    </Button>
    <Button
      variant="text"
      color="primary"
      href="contact"
    >
      Contact
    </Button>
  </>
);
interface DashboardProps {
  userState?: UserState;
}
