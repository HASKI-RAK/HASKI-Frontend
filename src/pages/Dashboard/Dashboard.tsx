import { AuthContext } from "@services/*";
import { useContext } from "react"
import { useNavigate } from "react-router-dom";

const Dashboard = () => {
    const authcontext = useContext(AuthContext);
    const navigate = useNavigate();
    return (
        authcontext.isAuth ?
            <div>
                Hello there!
                General Kenobi! You are a bold one!
                To home:
                <button onClick={() => navigate('/')}>Home</button>
                <br />
                <button onClick={() => authcontext.setIsAuth(false)}>Logout</button>
            </div>
            :
            <div>
                You are not authorized to view this page.
            </div>
    )
};

export default Dashboard;
