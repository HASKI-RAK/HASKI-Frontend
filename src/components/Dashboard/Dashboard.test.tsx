import { fireEvent, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import { Dashboard } from "./Dashboard";


global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve({ status: 200 }),
  }),
) as jest.Mock;

describe("Test the demo component", () => {
  const mockCallback = jest.fn();

  test("renders HASKI basic label", () => {
    const landingPage = render(
      <Dashboard
        userState={{
          user: { firstName: "Max", surName: "Mustermann" },
          increaseUserId: () => mockCallback(),
        }}
      />
    );
    const button = landingPage.getByText("one up");
    fireEvent.click(button);
    expect(mockCallback.mock.calls.length).toEqual(1);
    expect(mockCallback).toBeCalled();
  });

  test("renders correctly", () => {
    const mockCallback = jest.fn();
    const tree = renderer
      .create(
        <Dashboard
          userState={{
            user: { firstName: "Max", surName: "Mustermann" },
            increaseUserId: () => mockCallback(),
          }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
