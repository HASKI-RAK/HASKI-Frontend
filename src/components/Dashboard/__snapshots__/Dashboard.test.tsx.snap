// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the demo component renders correctly 1`] = `
[
  <div>
    Hello 
    Max
     
    Mustermann
     with ID:
  </div>,
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    one up
  </button>,
  <a
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    href="contact"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
  >
    Contact
  </a>,
]
`;
